tiledb/sm/group/group.cc

```diff
@@ -530,6 +532,28 @@ void Group::set_metadata_loaded(const bool metadata_loaded) {
   metadata_loaded_ = metadata_loaded;
 }
 
+void Group::vacuum_metadata(
+    ContextResources& resources, const char* group_name, const Config& config) {
+  // Check group URI
+  URI group_uri(group_name);
+  if (group_uri.is_invalid()) {
+    throw GroupException("Cannot vacuum group metadata; Invalid URI");
+  }
+
+  // Check if group exists
+  ObjectType obj_type;
+  throw_if_not_ok(object_type(resources, group_uri, &obj_type));
+
+  if (obj_type != ObjectType::GROUP) {
+    throw GroupException("Cannot vacuum group metadata; Group does not exist");
+  }
+
+  StorageManager storage_manager(resources, resources.logger(), config);
```

Comment:
Needing to make a new storage manager just to call `Consolidator::create` is quite a red flag. The problem isn't in this code narrowly, though; it's the implementation of the consolidator.

We should target changing `Consolidator::create` for a refactor. It should be possible to convert it to `ContextResources` and `RestClient`, but it will also entail changing a constructor signature for `Array`. It's definitely going to require a concerted effort. Removing all the array and group functions from `StorageManager` should happen first.

Reply:
There's already a followup story to remove usage of `StorageManager` from the consolidation code: https://app.shortcut.com/tiledb-inc/story/47947/eliminate-storagemanager-arguments-in-consolidation-code-in-favor-of-contextresources